components/reusable-components/blogDetails-card/BlogDetailsCard.tsx

```diff
@@ -0,0 +1,305 @@
+        {validationErrors.image && <p className={styles.errorText}>{validationErrors.image}</p>}
+        {isEditable ? (
+          <div {...getRootProps()} className={styles.dropzone}>
+            <input {...getInputProps()} id="imageUpload" />
+            <p>Drag 'n' drop some files here, or click to select files</p>
+            {acceptedFiles.length > 0 && (
+              <ul>
+                {acceptedFiles.map((file) => (
+                  <li key={file.name}>{file.name}</li>
+                ))}
+              </ul>
+            )}
+            {rejectedFiles.length > 0 && (
+              <ul>
+                {rejectedFiles.map(({ file, errors }) => (
+                  <li key={file.name}>
+                    {file.name} - {errors.map((e) => e.message).join(', ')}
+                  </li>
+                ))}
+              </ul>
+            )}
+          </div>
+        ) : (
+          imageUrl && <Image src={imageUrl} alt="Blog" width={400} height={300} />
+        )}
+      </div>
+
+      <div className={styles.contentSection}>
+        <label htmlFor="content">Content:</label>
+        {isEditable ? (
+          <TiptapEditor
+            content={content}
+            editable={isEditable}
+            onChange={(editorContent: string) =>
+              onChange({ target: { name: 'content', value: editorContent } })
+            }
+          />
+        ) : (
+          <div dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(content) }} />
+        )}
+      </div>
+
+      <div className={styles.authorSection}>
+        <label htmlFor="authorSelect">Select Author:</label>
+        <select
+          id="authorSelect"
+          value={selectedAuthorId}
+          onChange={handleAuthorChange}
+          disabled={!isEditable}
+          className={styles.inputField}
+        >
+          <option value="" disabled>
+            Select an author
+          </option>
+          {hardcodedAuthors.map((author) => (
+            <option key={author.id} value={author.id}>
+              {author.first_name} {author.last_name}
+            </option>
+          ))}
+        </select>
+
+        <label htmlFor="authorFirstName">Author First Name:</label>
+        <input
+          id="authorFirstName"
+          type="text"
+          value={propAuthor.first_name}
+          onChange={onChange as (event: React.ChangeEvent<HTMLInputElement>) => void}
+          name="author_first_name"
+          disabled={!isEditable}
+          className={styles.inputField}
+          required
+          placeholder="First name is required"
+        />
+        <label htmlFor="authorLastName">Author Last Name:</label>
+        <input
+          id="authorLastName"
+          type="text"
+          value={propAuthor.last_name}
+          onChange={onChange as (event: React.ChangeEvent<HTMLInputElement>) => void}
+          name="author_last_name"
+          disabled={!isEditable}
+          className={styles.inputField}
+          required
```

Comment:
The author first and last name inputs are not necessary.

Reply:
Should I add just one input: Author ?

Reply:
Yes we can have an input for Author. It will be disabled regardless if we are in Edit Mode or not.